# Joint state controller
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

# Gazebo-specific PID gains
gazebo_ros_control:
  pid_gains:
    # Right arm
    right_shoul_base2shoul_joint: {p: 100, i: 1, d: 0.001}
    right_shoul2shoul_rot_joint: {p: 100, i: 1, d: 0.1}
    right_arm2armrot_joint: {p: 100, i: 1, d: 0.001}
    right_armrot2elbow_joint: {p: 100, i: 1, d: 0.001}
    right_forearm2forearmrot_joint: {p: 100, i: 1, d: 0.001}
    right_forearmrot2forearm_pitch_joint: {p: 100, i: 1, d: 0.001}
    right_forearm_pitch2forearm_roll_joint: {p: 100, i: 1, d: 0.001}
    # Left arm
    left_shoul_base2shoul_joint: {p: 100, i: 1, d: 0.001}
    left_shoul2shoul_rot_joint: {p: 100, i: 1, d: 0.1}
    left_arm2armrot_joint: {p: 100, i: 1, d: 0.001}
    left_armrot2elbow_joint: {p: 100, i: 1, d: 0.001}
    left_forearm2forearmrot_joint: {p: 100, i: 1, d: 0.001}
    left_forearmrot2forearm_pitch_joint: {p: 100, i: 1, d: 0.001}
    left_forearm_pitch2forearm_roll_joint: {p: 100, i: 1, d: 0.001}

# Right arm position controller
right_arm_position_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - right_shoul_base2shoul_joint
    - right_shoul2shoul_rot_joint
    - right_arm2armrot_joint
    - right_armrot2elbow_joint
    - right_forearm2forearmrot_joint
    - right_forearmrot2forearm_pitch_joint
    - right_forearm_pitch2forearm_roll_joint
  
  constraints:
    goal_time: 0.5
    stopped_velocity_tolerance: 0.02
    right_shoul_base2shoul_joint:
      trajectory: 1.0
      goal: 0.5
    right_shoul2shoul_rot_joint:
      trajectory: 1.0
      goal: 0.5
    right_arm2armrot_joint:
      trajectory: 1.0
      goal: 0.5
    right_armrot2elbow_joint:
      trajectory: 1.0
      goal: 0.5
    right_forearm2forearmrot_joint:
      trajectory: 1.0
      goal: 0.5
    right_forearmrot2forearm_pitch_joint:
      trajectory: 1.0
      goal: 0.5
    right_forearm_pitch2forearm_roll_joint:
      trajectory: 1.0
      goal: 0.5

  gains:
    right_shoul_base2shoul_joint: {p: 100, i: 1, d: 0.001}
    right_shoul2shoul_rot_joint: {p: 100, i: 1, d: 0.1}
    right_arm2armrot_joint: {p: 100, i: 1, d: 0.001}
    right_armrot2elbow_joint: {p: 100, i: 1, d: 0.001}
    right_forearm2forearmrot_joint: {p: 100, i: 1, d: 0.001}
    right_forearmrot2forearm_pitch_joint: {p: 100, i: 1, d: 0.001}
    right_forearm_pitch2forearm_roll_joint: {p: 100, i: 1, d: 0.001}

# Left arm position controller
left_arm_position_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - left_shoul_base2shoul_joint
    - left_shoul2shoul_rot_joint
    - left_arm2armrot_joint
    - left_armrot2elbow_joint
    - left_forearm2forearmrot_joint
    - left_forearmrot2forearm_pitch_joint
    - left_forearm_pitch2forearm_roll_joint

  constraints:
    goal_time: 0.5
    stopped_velocity_tolerance: 0.02
    left_shoul_base2shoul_joint:
      trajectory: 1.0
      goal: 0.5
    left_shoul2shoul_rot_joint:
      trajectory: 1.0
      goal: 0.5
    left_arm2armrot_joint:
      trajectory: 1.0
      goal: 0.5
    left_armrot2elbow_joint:
      trajectory: 1.0
      goal: 0.5
    left_forearm2forearmrot_joint:
      trajectory: 1.0
      goal: 0.5
    left_forearmrot2forearm_pitch_joint:
      trajectory: 1.0
      goal: 0.5
    left_forearm_pitch2forearm_roll_joint:
      trajectory: 1.0
      goal: 0.5

  gains:
    left_shoul_base2shoul_joint: {p: 100, i: 1, d: 0.001}
    left_shoul2shoul_rot_joint: {p: 100, i: 1, d: 0.1}
    left_arm2armrot_joint: {p: 100, i: 1, d: 0.001}
    left_armrot2elbow_joint: {p: 100, i: 1, d: 0.001}
    left_forearm2forearmrot_joint: {p: 100, i: 1, d: 0.001}
    left_forearmrot2forearm_pitch_joint: {p: 100, i: 1, d: 0.001}
    left_forearm_pitch2forearm_roll_joint: {p: 100, i: 1, d: 0.001}